pandas/core/internals/blocks.py

```diff
@@ -1866,6 +1840,36 @@ def _unstack(self, unstacker, fill_value, new_placement):
         ]
         return blocks, mask
 
+    def quantile(self, qs, interpolation="linear", axis: int = 0) -> Block:
+        # asarray needed for Sparse, see GH#24600
+        mask = np.asarray(isna(self.values))
+        mask = np.atleast_2d(mask)
+
+        values, fill_value = self.values._values_for_factorize()
```

Comment:
In general, `_values_for_factorize` is not necessarily correct, but since quantile only makes sense for numeric-like dtypes (and for those dtypes, `_values_for_factorize` will most likely be the underlying numbers), that's probably OK.

(eg in geopands, `_values_for_factorize` returns serialized bytes, but quantile/percentile doesn't work on bytes anyway, so no direct problem)

Reply:
i think quantile makes sense for any ordered types, and IIUC values_for_factorize is supposed to preserve ordering

Reply:
> IIUC values_for_factorize is supposed to preserve ordering

In *theory* I think that's not required for factorize (since we don't sort in factorization). In practice not sure if that matters. 
But looking at my overview in https://github.com/pandas-dev/pandas/issues/33276 again, I mentioned there that in principle we shouldn't use `_values_for_factorize` directly, but rather use the `.factorize()` method. But of course that's not usable for this use case.